### PR TITLE
Use MTF for tests

### LINKED ISSUE #45: Move tests to MTF
After investigating the current tests, we concluded that it would be good to move them to the MorphicTestFramework (MTF).

### PR DESCRIPTION
**Pull request by ClFeSc**
Closes #45

### CODE REVIEW

**Comment by ClFeSc** [comment]
My plan is to first move all tests to the new class. I already marked some tests as questionable, feel free to point out more such tests.

File: packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testHasColorPaletteEnoughColors.st [hunk 1]
```diff
@@ -1,5 +1,6 @@
 UI existence tests
 testHasColorPaletteEnoughColors
+"Do we want to keep this test?"
 	| palette |
-	palette := self morphicMonet colorPalette.
+	palette := self subject colorPalette.
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testHasColorPaletteEnoughColors.st]: In my opinion the test is not necessary anymore.

File: packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testHasColorPaletteNoDuplicates.st [hunk 2]
```diff
@@ -1,7 +1,8 @@
 UI existence tests
 testHasColorPaletteNoDuplicates
+"Do we want to keep this test?"
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testHasColorPaletteNoDuplicates.st]: In my opinion the test is not necessary anymore.

**Review by maximilian-franz — COMMENTED** [review]
Some files have two dots before the file ending, e.g. [baseline..st](https://github.com/hpi-swa-teaching/SketchMorph2/blob/1dd43fe91ab19732a304746d445927fff335bd99/packages/BaselineOfSketchMorph2.package/BaselineOfSketchMorph2.class/instance/baseline..st). Also, should we insert a newline at the end of a .st file (as suggested by github)?

Other than that, everything looks fine.

**Comment by ClFeSc** [comment]
@maximilian-franz 
> Some files have two dots before the file ending, e.g. [baseline..st](https://github.com/hpi-swa-teaching/SketchMorph2/blob/1dd43fe91ab19732a304746d445927fff335bd99/packages/BaselineOfSketchMorph2.package/BaselineOfSketchMorph2.class/instance/baseline..st). 

Sure. What's the problem with them?
> Also, should we insert a newline at the end of a .st file (as suggested by github)?

I agree. I will add them to all files of the `MorphicMonetTests` class, the other files are out of scope of this PR, I think.

**Comment by maximilian-franz** [comment]
@ClFeSc 

> Sure. What's the problem with them?

It's not necessarily a problem, it just looks a bit weird (at least to me).

**Comment by ClFeSc** [comment]
@maximilian-franz 
> It's not necessarily a problem, it just looks a bit weird (at least to me).

I'm not sure if you know this, but Squeak uses `.` in file names wherever a `:` in the method signature is. So the file for the method `baseline:` will get the name `baseline.`, which is then appended to `.st`, thus creating `baseline..st`. I don't think there is much we can do about it.

**Comment by maximilian-franz** [comment]
@ClFeSc 
> I'm not sure if you know this, but Squeak uses `.` in file names wherever a `:` in the method signature is. So the file for the method `baseline:` will get the name `baseline.`, which is then appended to `.st`, thus creating `baseline..st`. I don't think there is much we can do about it.

Ah ok, I didn't know that.

File: packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/isButtonThere..st [hunk 3]
```diff
@@ -1,3 +1,3 @@
 support
 isButtonThere: aLabel
-	^ ((self lookForMenuWithItem: aLabel) isNil) not
\ No newline at end of file
+	^ ((self lookForMenuWithItem: aLabel) isNil) not
```
Comment (DevSchmidtchen) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/isButtonThere..st]: `isNil not` should be equals to `notNil` and could be replaced.

**Review by DevSchmidtchen — CHANGES_REQUESTED** [review]

File: packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testHasColorPaletteEvenAmountOfColors.st [hunk 4]
```diff
@@ -1,6 +1,7 @@
 UI existence tests
 testHasColorPaletteEvenAmountOfColors
+"Do we want to keep this test?"
```
Comment (kpostnov) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testHasColorPaletteEvenAmountOfColors.st]: This one can be removed as well.

File: packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKey.modifiedBy..st [hunk 5]
```diff
@@ -0,0 +1,3 @@
+*SketchMorph2-Tests
+sendKey: aCharacter modifiedBy: aModifier
+	self sendKeyEvent: #keystroke characterValue: aCharacter asInteger modifiedBy: aModifier.
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKey.modifiedBy..st]: Final newline is missing.

File: packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/doButtonAction..st [hunk 6]
```diff
@@ -1,6 +1,8 @@
...
-	submenu := self lookForMenuWithItem: aLabel.
-	button := submenu submorphs detect: [:item | item contents = aLabel].
-	button doButtonAction
\ No newline at end of file
+	| morphs |
+	morphs := self subject findByCriteria: [:aMorph |
+		(aMorph respondsTo: #contents) and: [aMorph contents = aLabel]].
+	self assert: morphs morphs size equals: 1.
+	morphs morphs do: [:aMorph |
+		aMorph doButtonAction]
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/doButtonAction..st]: Final newline is missing.

Comment (johannes-poetzsch) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/doButtonAction..st]: missing newline

**Review by maximilian-franz — CHANGES_REQUESTED** [review]

File: packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKey.modifiedBy..st (hunk 5 above)
Comment (johannes-poetzsch) [inline packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKey.modifiedBy..st]: missing newLine

Comment (maximilian-franz) [reply packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKey.modifiedBy..st]: Also final dot

File: packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKeyEvent.characterValue.modifiedBy..st [hunk 7]
```diff
@@ -0,0 +1,7 @@
+*SketchMorph2-Tests
+sendKeyEvent: aType characterValue: aValue modifiedBy: aModifier
+	
+	self morphs do: [ :morph |
+		self hand newKeyboardFocus: morph.
+		morph handleEvent: (self generateKeyboardEvent: aType with: aValue on: morph modifiedBy: aModifier).
+		self wait].
```
Comment (johannes-poetzsch) [inline packages/SketchMorph2-Tests.package/MTFMorphWrapper.extension/instance/sendKeyEvent.characterValue.modifiedBy..st]: final dot should be removed

File: packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testPasteClipboardImage.st [hunk 8]
```diff
@@ -1,6 +1,6 @@
 mockObject tests
 testPasteClipboardImage
 	Clipboard clipboardText: self writeTestPNG.
-	self hand newKeyboardFocus: self morphicMonet.
-	self hand handleEvent: self generatePasteEvent.
-	self assert: self morphicMonet backend lastCalledMethod == #importPNG:
\ No newline at end of file
+	self subject sendKey: $v modifiedBy: MTFMorphWrapper commandModifier.
+	self assert: self subject backend lastCalledMethod == #importPNG:
+
```
Comment (johannes-poetzsch) [inline packages/SketchMorph2-Tests.package/MorphicMonetTests.class/instance/testPasteClipboardImage.st]: there seems to be an additional new line at the end that should be removed

**Review by maximilian-franz — APPROVED** [review]

**Review by kpostnov — APPROVED** [review]

**Review by johannes-poetzsch — APPROVED** [review]

**Review by jannis-hpi — APPROVED** [review]

**Review by DevSchmidtchen — APPROVED** [review]

**Pull request merged**